src/Polysemy/Fish.hs

```diff
@@ -0,0 +1,282 @@
+-- "Fish" operators ----------------------------------------------------------
+------------------------------------------------------------------------------
+-- | Infix equivalent of 'Sem' with automatic conversion of supplied effects
+-- into list. Use ('>@>') instead if you only want to work with 'Members' of
+-- your union.
+--
+-- __Examples:__
+--
+-- 'Sem' with list of multiple effects:
+--
+-- @
+-- foo :: 'Sem' \'['Polysemy.State.State' 'Int', 'Polysemy.Input.Input' 'String'] ()
+-- @
+--
+-- can be written as:
+--
+-- @
+-- foo :: \'['Polysemy.State.State' 'Int', 'Polysemy.Input.Input' 'String'] \@> ()
+-- @
+--
+-- 'Sem' with list of one effect:
+--
+-- @
+-- foo :: 'Sem' \'['Polysemy.State.State' 'Int'] ()
+-- @
+--
+-- can be written as both:
+--
+-- @
+-- foo :: \'['Polysemy.State.State' 'Int'] \@> ()
+-- @
+--
+-- and:
+--
+-- @
+-- foo :: 'Polysemy.State.State' 'Int' \@> ()
+-- @
+--
+-- where effect without list gets put into one automatically.
+--
+-- 'Sem' with __exactly__ one, lifted monad:
+--
+-- @
+-- foo :: 'Sem' \'['Lift' 'IO'] ()
+-- @
+--
+-- can be written simply as:
+--
+-- @
+-- foo :: 'IO' \@> ()
+-- @
+--
+-- and will be automatically lifted and put into list.
+--
+-- __NOTE:__ For this operator to work, you have to provide types of monad and
+-- contained value in your effect:
+--
+-- @
+-- data MyEffect x y (m :: * -> *) (a :: *) where ...
+-- @
+--
+-- but this requirement may be removed in the future.
+infix 2 @>
+type es @> r = Sem (ToEffList "(@>)" es) r
+
+-- | Infix equivalent of 'Members' constraint used directly in /return/ type,
+-- with automatic conversion of supplied effects into list. Can be used in
+-- combination with ('@>'). Use ('>@>') instead if you only want to work with
+-- 'Members' of your union.
+--
+-- __Examples:__
+--
+-- List of multiple members:
+--
+-- @
+-- foo :: 'Members' \'['Polysemy.State.State' 'Int', 'Polysemy.Input.Input' 'String'] r => 'Sem' (ConsoleIO:r) () -> 'Sem' r ()
+-- @
+--
+-- can be written as:
+--
+-- @
+-- foo :: ConsoleIO:r \@> () -> \'['Polysemy.State.State' 'Int', 'Polysemy.Input.Input' 'String'] >\@ r \@> ()
+-- @
+--
+-- One member:
+--
+-- @
+-- foo :: 'Member' ('Polysemy.State.State' 'Int') r => 'Sem' (ConsoleIO:r) () -> 'Sem' r ()
+-- @
+--
+-- can be written as both:
+--
+-- @
+-- foo :: ConsoleIO:r \@> () -> \'['Polysemy.State.State' 'Int'] >\@ r \@> ()
+-- @
+--
+-- and:
+--
+-- @
+-- foo :: ConsoleIO:r \@> () -> 'Polysemy.State.State' 'Int' >\@ r \@> ()
+-- @
+--
+-- where effect without list gets put into one automatically.
+--
+-- __Exactly__ one, lifted monad as a member:
+--
+-- @
+-- foo :: 'Member' ('Lift' 'IO') r => 'Sem' (ConsoleIO:r) () -> 'Sem' r ()
+-- @
+--
+-- can be written simply as:
+--
+-- @
+-- foo :: ConsoleIO:r \@> () -> 'IO' >\@ r \@> ()
+-- @
+--
+-- and will be automatically lifted and put into list.
+--
+-- __NOTE:__ For this operator to work, you have to provide types of monad and
```

Comment:
"you need to provide an explicit `* -> *` kind signature on the `m` parameter"

Reply:
I can look at convincing the plugin to provide these kind sigs too; I don't think it would be very hard.